pytensor/link/jax/dispatch/tensor_basic.py

```diff
@@ -205,3 +211,39 @@ def tri(*args):
         return jnp.tri(*args, dtype=op.dtype)
 
     return tri
+
+
+@node_rewriter([AllocDiag])
+def eagerly_inline_alloc_diag(fgraph, node):
+    """
+    Inline `AllocDiag` OpFromGraph into the graph so the component Ops can themselves be jaxified
+
+    Parameters
+    ----------
+    fgraph: FunctionGraph
+        The function graph being rewritten
+    node: Apply
+        Node of the function graph to be optimized
+
+    Returns
+    -------
+
+    """
+    [input] = node.inputs
+    [output] = node.op.inner_outputs
+    inner_input = output.owner.inputs[1]
+
+    inline = pytensor.clone_replace(output, {inner_input: input})
+
+    return [inline]
```

Comment:
We should call a helper function that inlines any OpFromGraph (i.e., the logic inside the original OFG inline), because we will probably want to control when we inline different OpFromGraph.

So the rewrite should be a parametrized one:

```python
def inline_ofg_subclass_rewrite(ofg_type):

  @node_rewriter(tracks=[ofg_type]):
  def inline_ofg_rewriter(fgraph, node):
      ... # logic that is independent of number of inputs/outputs (also doesn't forget copy_stack_trace)

  return inline_ofg_rewriter
```

Reply:
Where should the helper live? A new `pytensor/builder/rewrite` module? `tensor/rewriting/basic`?